### DISCUSSION TITLE
compileKotlinIosSimulatorArm64 FAILED - kmmp using supabase works for android, failed dep. for ios

### DISCUSSION THREAD

**Discussion by MichaelJob** [supabase-community/supabase-kt#490] [Q&A] [answered]
how can I resolve:
Task :shared:compileKotlinIosSimulatorArm64 FAILED
warning: skipping /Users/mjob/.gradle/caches/modules-2/files-2.1/io.github.jan-tennert.supabase/postgrest-kt-iossimulatorarm64/2.1.5/3006377e5bbc40389b69f2e20f05090f057aa4a0/postgrest-kt.klib. Incompatible abi version. The current default is '1.7.0', found '1.8.0'. The library produced by 1.9.22 compiler
error: Could not find "/Users/mjob/.gradle/caches/modules-2/files-2.1/io.github.jan-tennert.supabase/postgrest-kt-iossimulatorarm64/2.1.5/3006377e5bbc40389b69f2e20f05090f057aa4a0/postgrest-kt.klib" in [/Users/mjob/AndroidStudioProjects/my-bird-app-main/iosApp/Pods, /Users/mjob/.konan/klib, /Users/mjob/.konan/kotlin-native-prebuilt-macos-aarch64-1.8.20/klib/common, /Users/mjob/.konan/kotlin-native-prebuilt-macos-aarch64-1.8.20/klib/platform/ios_simulator_arm64]

I tried with the older 1.4.7 and newer 2.1.3 and 2.2.0-rc-1 dependency in commonMain, with no luck.

```           
implementation(platform("io.github.jan-tennert.supabase:bom:2.1.5"))
implementation("io.github.jan-tennert.supabase:postgrest-kt")
implementation("io.github.jan-tennert.supabase:storage-kt")
```

**Comment by jan-tennert** [comment]
@MichaelJob It looks like some weird gradle cache problem. Can you try running `gradlew clean` in your project folder and delete the `.gradle` folder in /Users/mjob

**Reply by MichaelJob** [reply]
did that, now all is broken - sync fails with loads of errors, like ```Failed building KotlinMPPGradleModel
org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreationException: Could not create task ':androidApp:compileDebugJavaWithJavac'.....```

**Reply by jan-tennert** [reply]
yea you probably have to restart/invalidate Android Studio's / IntelliJ caches

**Comment by jan-tennert** [comment]
Also, which Kotlin version do you use? If not 1.9.X, this might also be a problem

**Reply by MichaelJob** [reply]
```
#Gradle
org.gradle.jvmargs=-Xmx2048M -Dkotlin.daemon.jvm.options\="-Xmx2048M"

#Kotlin
kotlin.code.style=official

#MPP
kotlin.mpp.stability.nowarn=true
kotlin.mpp.enableCInteropCommonization=true
kotlin.mpp.androidSourceSetLayoutVersion=2

#Compose
org.jetbrains.compose.experimental.uikit.enabled=true
kotlin.native.cacheKind=none

#Android
android.useAndroidX=true
android.compileSdk=34
android.targetSdk=33
android.minSdk=24

#Versions
kotlin.version=1.8.20
agp.version=8.0.0
compose.version=1.4.0
```
tried with Kotlin 1.9.22 and compose 1.5.12 and changed a few recommondations of all the errors, but to no success - reverted back

**Reply by MichaelJob** [reply]
the repo is here: https://github.com/MichaelJob/kmpBicylcles

**Reply by jan-tennert** [reply]
Don't have a MAC, so unfortunately can't test it myself

**Reply by MichaelJob** [reply]
thank you for your efforts anyways

**Reply by MichaelJob** [reply, marked as answer]
@jan-tennert after update to Android Studio Iguana and trying again with the updates, it now works on Android and iOS - hooray - 
now with:
```
#Versions in gradle.properties
kotlin.version=1.9.22
agp.version=8.2.0
compose.version=1.5.12

//Supabase in build.gradle.kts (shared):
implementation(project.dependencies.platform("io.github.jan-tennert.supabase:bom:2.1.5"))
implementation("io.github.jan-tennert.supabase:postgrest-kt")
implementation("io.github.jan-tennert.supabase:storage-kt")
```
